articles/send-sms-in-nodejs-using-vonage-api/index.md

```diff
@@ -0,0 +1,229 @@
+```js
+// phone number
+const phoneNumber= document.getElementById('number'),
+
+//message to send
+textMessage = document.getElementById('message'),
+
+//send button
+button = document.getElementById('button'),
+
+//response field
+response = document.querySelector(".respose");
+```
+
+### Adding the send functionality
+since we have grabbed our send button from the html form, we need to add an even listener to the button to trigger the send message function. The send fucntionality code is as shown  below:
+```js
+button.addEventListener('click', send, false);
+function send() {
+    // getting value form the variabled  fetched
+    const number = phoneNumber.value.replace(/\D/g, '');// elim inate non numeric characters from out number
+    const text = textMessage.value;
+    
+    // Making post request with the data from our server
+    fetch('/', {
+        method: 'post',
+        headers: {
+          'Content-type': 'application/json'
+        },
+        // convert the request body into a string
+        body: JSON.stringify({ number, text })
+    }).then(function (res){
+        //log the response
+        console.log(res);
+    }).catch(function(err){ //cathc any error found
+        //console log the error
+        console.log(err);
+    });
+}
+```
+#### Implemeting Nexmo
+First we will initialize Nexmo by supplyoing out apiKey and secret
+```js
+//init out nexmo
+const nexmo = new Nexmo({
+    apiKey: 'YOUR API-KEY',
+    apiSecret: 'API-SECRET',
+}, {debug:true});
+```
+We need to catch the post on the server since the fetch API makes a post request. In our `app.js`,  we will add the following code to catch the post request and use Nexmo to send the message to the number specified in the request body.
+```js
+//catch post from our main js
+app.post('/', (request, response) =>{
+    const phoneNumber = request.body.number;
+    const textMessage = request.body.text;
+
+    const from = 'Vonage APIs';
+    const to = phoneNumber;
+    const text =  textMessage;
+    nexmo.message.sendSms(from, to, text, {type:'unicode'}, (error,responseData) =>{
+        if(error){
+            console.log(error)
+        }else{
+            console.dir(responseData);
+            //get phone number into an object to be sent to the client
+            const data = {
+                number: responseData.messages[0]['to']
+            }
+            //send response to client using socket 
+            io.emit('smsStatus', data);
+        }
+    });
+});
+```
+Next, we will send the repose data to the client to show if the SMS was sent successfully or not. To do this, we will send the response data from nexmo to the client in the `main.js` file using `io.emit()` function. Ypu could find more about this fucntion [here](https://socket.io/docs/v3/).
```

Comment:
Next, we will send the `repose` data to the client to show if the SMS was sent successfully or not.

Do you mean response